File: ios/DeviceIdModule.m
//
//  DeviceIdModule.m
//  mirrorfly_rn
//
//  Created by user on 24/04/25.

// DeviceIdModule.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>


@interface RCT_EXTERN_MODULE(DeviceId, NSObject)
RCT_EXTERN_METHOD(getDeviceID:(RCTPromiseResolveBlock *)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: android/app/src/main/java/com/mirrorfly_rn/DeviceIdModule.kt
package com.mirrorfly_rn

import android.annotation.SuppressLint
import android.provider.Settings
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactMethod

class DeviceIdModule(private val reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName(): String = "DeviceId"

    @SuppressLint("HardwareIds")
    @ReactMethod
    fun getDeviceID(promise: Promise) {
        try {
            val androidId = Settings.Secure.getString(
                reactContext.contentResolver,
                Settings.Secure.ANDROID_ID
            )
            promise.resolve(androidId)
        } catch (e: Exception) {
            promise.reject("DEVICE_ID_ERROR", "Failed to get Android ID", e)
        }
    }
}
